src/formatDate.js

```diff
@@ -8,7 +8,76 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const formatDictionary = {};
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const inputDateParts = date.split(oldSeparator);
+  const newDateParts = [];
+
+  /* Create a dictionary with:
+   * - key - first letter of the part format ('D', 'M' or 'Y')
+   * - value - dictionary with the given 'format' and 'value' of this date part
+   */
+  for (let i = 0; i < 3; i++) {
+    const format = fromFormat[i];
+    const formatCategory = format[0];
+    const value = inputDateParts[i];
+
+    formatDictionary[formatCategory] = {
+      format: format,
+      value: value,
+    };
```

Comment:
The `formatDictionary` is being populated inside a loop, which is not necessary since you are only using it to map the format to its corresponding value. Consider restructuring this to avoid the use of an object as a dictionary when a simpler data structure like an array could be used more effectively.